tools/go-cli/go-whisk/whisk/wskerror.go

```diff
@@ -25,64 +25,141 @@ const EXITCODE_ERR_USAGE        int = 2
-//  bool    - DisplayMsg.  If true, the error message should be displayed on the console
-//  bool    - DisplayUsage.  If true, the command usage syntax/help should be displayed on the console
-//  bool    - MsgDisplay.  If true, the error message has been displayed on the console
-func MakeWskErrorFromWskError (e error, werr error, ec int, flags ...bool ) (we *WskError) {
-    var exitCode int = ec
-
-    // Pull exit code from wsk error if it exists
-    if werr !=nil {
-        // werr can either be a pointer reference
-        if we, ok := werr.(*WskError); ok {
-            exitCode = we.ExitCode
-            flags[1] = we.DisplayUsage || flags[1]  // Display usage syntax if either error requests it
-        }
+/*
+Instantiate a WskError structure
+Parameters:
+    error       - RootErr. object implementing the error interface
+    WskError    - WskError being wrappered.  It's exitcode will be used as this WskError's exitcode.  Ignored if nil
+    int         - ExitCode.  Used if error object is nil or if the error object is not a WskError
+    bool        - DisplayMsg.  If true, the error message should be displayed on the console
+    bool        - DisplayUsage.  If true, the command usage syntax/help should be displayed on the console
+    bool        - MsgDisplayed.  If true, the error message has been displayed on the console
+    bool        - DisplayPrefix.  If true, the error message will be prefixed with "error: "
+*/
+func MakeWskErrorFromWskError (baseError error, whiskError error, exitCode int, flags ...bool) (resWhiskError *WskError) {
 
-        // ... or a value
-        if we, ok := werr.(WskError); ok {
-            exitCode = we.ExitCode
-            flags[1] = we.DisplayUsage || flags[1]  // Display usage syntax if either error requests it
+    // Get the exit code, and flags from the existing Whisk error
+    if whiskError != nil {
+
+        // Ensure the Whisk error is a pointer
+        switch errorType := whiskError.(type) {
+            case *WskError:
+                resWhiskError = errorType
+            case WskError:
+                resWhiskError = &errorType
         }
+
+        exitCode, flags = getWhiskErrorProperties(resWhiskError, flags...)
+    }
+
+    return MakeWskError(baseError, exitCode, flags...)
+}
+
+/*
+Returns the settings from a WskError. Values returned will include ExitCode, DisplayMsg, DisplayUsage, MsgDisplayed,
+and DisplayPrefix.
+
+Parameters:
+    whiskError  - WskError to examine.
+    flags       - Boolean values that may override the WskError object's values for DisplayMsg, DisplayUsage,
+                    MsgDisplayed, and DisplayPrefix.
+ */
+func getWhiskErrorProperties(whiskError *WskError, flags ...bool) (int, []bool) {
+    if len(flags) > 0 {
+        flags[0] = whiskError.DisplayMsg
+    } else {
+        flags = append(flags, whiskError.DisplayMsg)
+    }
+
+    if len(flags) > 1 {
+        flags[1] = whiskError.DisplayUsage || flags[1]
```

Comment:
Is flags[1] supposed to override (per function comment) the whiskError.DisplayUsage value?  It won't if it is false and whiskError.DisplayUsage is true.